pkg/reconciler/apis/apiexportendpointslice/apiexportendpointslice_reconcile.go

```diff
@@ -64,35 +68,103 @@ func (r *endpointsReconciler) reconcile(ctx context.Context, apiExportEndpointSl
 	}
 	apiExport, err := r.getAPIExport(apiExportPath, apiExportEndpointSlice.Spec.APIExport.Name)
 	if err != nil {
+		reason := apisv1alpha1.InternalErrorReason
 		if errors.IsNotFound(err) {
 			// Don't keep the endpoints if the APIExport has been deleted
 			apiExportEndpointSlice.Status.APIExportEndpoints = nil
+			// No need to try again
+			err = nil
+			reason = apisv1alpha1.APIExportNotFoundReason
```

Comment:
why not to set condition here and return early?

Reply:
The following code does not do anything else. It sets the conditions and returns.
I did not see the need to duplicate.